x-pack/plugins/observability_solution/dataset_quality/server/routes/register_routes.ts

```diff
@@ -86,6 +95,11 @@ export function registerRoutes({ repository, core, logger, plugins }: RegisterRo
             opts.body.message = 'Client closed request';
           }
 
+          if (error instanceof UnsupportedFeatureInOfferingError) {
+            opts.statusCode = 501;
```

Comment:
I don't want to overcomplicate this, but sending a 501 here might trigger some questions from SRE during production as the product is functioning as expected and it's not an actual serious problem (which 5xx normally indicates). What about returning the request as success, but adding the fact that we can't display the information in the payload and set the disabled flag based on that?

Something like
```ts
{
    canEstimateSize: true;
    estimatedDataInBytes: number;
} | {
    canEstimateSize: false;
}
```

as the response type.

Reply:
In another PR where I've to deal with the same scenario, I am [returning](https://github.com/elastic/kibana/pull/179479/files#diff-6fee27e4708336a5406ffb1c8e13d93db6a2e2e3056e2c42884d9aa7f7a83b6dR47) `NaN` for the size metric which is encoded as `null` in response payload. This indicates the frontend that this metric isn't available and the component isn't shown. The endpoint never returns `NaN` in any other scenario and it's only a temporary measure soon to be removed. Plus [API](https://github.com/elastic/kibana/pull/179479/files#diff-b90d31f99fa5d26ca427ebccacf2caf5030a49c0ed982ebac2940f458728eab3R91) and [e2e](https://github.com/elastic/kibana/pull/179479/files#diff-b90d31f99fa5d26ca427ebccacf2caf5030a49c0ed982ebac2940f458728eab3R91) tests cover this Serverless case.

See if it makes sense to do the same here.

Reply:
These APIs are not (yet) considered public, right? They are just used to power the UI? If yes I think it's OK to keep it like this for now and remove it once the new API is in - no need to overcomplicate for something that will be gone in a version.

Reply:
I like better Joe's approach because it's more explicit but as he already mentioned this is something that will be gone soon 🤞🏼

Reply:
Yes, these endpoints are only `/internal/` for now and feed the data to Dataset Quality UI.

Reply:
I would suggest going with @awahab07 s approach for now then 👍

Reply:
(Back from Easter break)

I'll change this to the `null` approach 👍

FWIW I don't think using a `501` is incorrect ("the server does not support the functionality required to fulfill the request" - and in a serverless offering this is the case), but if this will confuse SREs that's fair enough, especially being temporary.